src/parser/ident.rs

```diff
@@ -1,47 +1,37 @@
-use super::{ParseError, ParseResult};
+use super::ParseResult;
+use crate::expect_token;
 use crate::Parser;
 use crate::{ByteString, TokenKind};
 
 impl Parser {
     /// Expect an unqualified identifier such as Foo or Bar.
     pub(crate) fn ident(&mut self) -> ParseResult<ByteString> {
-        Ok(expect!(
-            self,
-            TokenKind::Identifier(i),
-            i,
-            "expected identifier"
-        ))
+        Ok(expect_token!([
+            TokenKind::Identifier(identifier) => identifier,
+        ], self, "an identifier"))
```

Comment:
i flipped the pattern and value on purpose, this now looks more like rust code :)

Reply:
yep, this is much nicer